composer/loggers/progress_bar_logger.py

```diff
@@ -57,50 +62,97 @@ def state_dict(self) -> Dict[str, Any]:
 
-    During training, the progress bar logs the batch and training loss.
-    During validation, the progress bar logs the batch and validation accuracy.
-
-    Example usage:
-        .. testcode::
-
-            from composer.loggers import ProgressBarLogger
-            from composer.trainer import Trainer
+    .. note::
 
-            trainer = Trainer(
-                ...,
-                loggers=[ProgressBarLogger()]
-            )
+        This logger is automatically instainatied by the trainer via the ``progress_bar``, ``log_to_console``,
+        ``log_level``, and ``console_stream`` options. This logger does not need to be created manually.
 
-        .. testcleanup::
+    `TQDM <https://github.com/tqdm/tqdm>`_ is used to display progress bars.
 
-            trainer.engine.close()
+    During training, the progress bar logs the batch and training loss.
+    During validation, the progress bar logs the batch and validation accuracy.
 
-    Example output::
+    Example progress bar output::
 
         Epoch 1: 100%|██████████| 64/64 [00:01<00:00, 53.17it/s, loss/train=2.3023]
         Epoch 1 (val): 100%|██████████| 20/20 [00:00<00:00, 100.96it/s, accuracy/val=0.0995]
 
-    .. note::
+    Args:
+        progress_bar (bool, optional): Whether to show a progress bar. (default: ``True``)
+        log_to_console (bool, optional): Whether to print logging statements to the console. (default: ``None``)
 
-        It is currently not possible to show additional metrics.
-        Custom metrics for the TQDM progress bar will be supported in a future version.
+            The default behavior (when set to ``None``) only prints logging statements when ``progress_bar`` is
+            ``False``.
+        console_log_level (LogLevel | str | (State, LogLevel) -> bool, optional): The maximum log level for which statements
+            should be printed. (default: :attr:`.LogLevel.EPOCH`)
+
+            It can either be :class:`.LogLevel`, a string corresponding to a :class:`.LogLevel`, or a callable that
+            takes the training :class:`.State` and the :class:`.LogLevel` and returns a boolean of whether this
+            statement should be printed.
+
+            This parameter has no effect if ``log_to_console`` is ``False`` or is unspecified when ``progress_bar`` is
+            ``True``.
+        stream (str | TextIO, optional): The console stream to use. If a string, it can either be ``'stdout'`` or
+            ``'stderr'``. (default: :attr:`sys.stderr`)
     """
 
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        progress_bar: bool = True,
+        log_to_console: Optional[bool] = None,
+        console_log_level: Union[LogLevel, str, Callable[[State, LogLevel], bool]] = LogLevel.EPOCH,
+        stream: Union[str, TextIO] = sys.stderr,
+    ) -> None:
+        self.show_pbar = progress_bar
+        log_to_console = not progress_bar if log_to_console is None else log_to_console
```

Comment:
Bit confused about this line, shouldn't the default be to show both the progress bar AND console logging? Wonder if this is related to Landan's comment above.